native/src/widget/grid.rs

```diff
@@ -0,0 +1,198 @@
+//! Write some text for your users to read.
+use crate::{
+    layout::{self, Node},
+    Element, Hasher, Layout, Length, Point, Size, Widget,
+};
+use std::iter;
+
+/// A container that distributes its contents in a grid.
+///
+/// # Example
+///
+/// ```
+/// # use iced_native::{renderer::Null, Element, Grid as NativeGrid, Text};
+/// # type Grid<'a> = NativeGrid<'a, (), Null>;
+/// Grid::from_vec(
+///     2,
+///     vec![
+///         Element::from(Text::new("First row, first column")),
+///         Element::from(Text::new("First row, second column")),
+///         Element::from(Text::new("Second row, first column")),
+///         Element::from(Text::new("Second row, second column")),
+///     ],
+/// );
+/// ```
+#[allow(missing_debug_implementations)]
+pub struct Grid<'a, Message, Renderer> {
+    columns: usize,
+    elements: Vec<Element<'a, Message, Renderer>>,
+}
+
+impl<'a, Message, Renderer> Grid<'a, Message, Renderer> {
+    /// Create a new empty [`Grid`].
+    ///
+    /// [`Grid`]: struct.Grid.html
+    pub fn new(columns: usize) -> Self {
+        Self::from_vec(columns, Vec::new())
+    }
+
+    /// Create a new [`Grid`] with the given [`Element`]s.
+    ///
+    /// [`Grid`]: struct.Grid.html
+    /// [`Element`]: ../struct.Element.html
+    pub fn from_vec(
+        columns: usize,
+        elements: Vec<Element<'a, Message, Renderer>>,
+    ) -> Self {
+        Self { columns, elements }
+    }
+
+    /// Adds an [`Element`] to the [`Grid`].
+    ///
+    /// [`Element`]: ../struct.Element.html
+    /// [`Grid`]: struct.Grid.html
+    pub fn push<E>(mut self, element: E) -> Self
+    where
+        E: Into<Element<'a, Message, Renderer>>,
+    {
+        self.elements.push(element.into());
+        self
+    }
+
+    /// Adds [`Element`]s to the [`Grid`].
+    ///
+    /// [`Element`]: ../struct.Element.html
+    /// [`Grid`]: struct.Grid.html
+    pub fn push_vec(
```

Comment:
For consistency, I wouldn't include this for now as no other widget has it. I am not sure this is something we want to encourage.

Reply:
Done.